### DISCUSSION TITLE
Warn about x-data inside x-for

### DISCUSSION THREAD

**Discussion by mariusGundersen** [alpinejs/alpine#1038] [4. General]
It seems like x-data inside an x-for template doesn't work correctly, but there are no warnings about this and at first glance it even appears to work correctly. 

Here is a simple example: https://codepen.io/mariusgundersen/pen/XWjGPMw

When the item in the list is changed the data isn't reinitialized. It looks like it is working on first render, but it doesn't behave correctly when it rerenders. For it to behave correctly the list has to be emptied and then populated again, so that the dom element is removed.

Can this either be fixed by implementing the functionality or by having a console warning/error that x-data inside x-for isn't supported?

**Comment by KevinBatdorf** [comment]
I'm not sure exactly what you're attempting to demo but I see a bug. Try using `:key="item"`
